test/e2e/prometheusagent_test.go

```diff
@@ -579,6 +580,57 @@ func testPrometheusAgentDaemonSetSelectPodMonitor(t *testing.T) {
 	require.NotEqual(t, firstTargetIP, secondTargetIP)
 }
 
+func testPrometheusAgentSSetServiceName(t *testing.T) {
+	t.Parallel()
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+	ns := framework.CreateNamespace(context.Background(), t, testCtx)
+	name := "test-agent-servicename"
+
+	svc := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-service", name),
+			Namespace: ns,
+		},
+		Spec: v1.ServiceSpec{
+			Type: v1.ServiceTypeLoadBalancer,
+			Ports: []v1.ServicePort{
+				{
+					Name: "web",
+					Port: 9090,
+				},
+			},
+			Selector: map[string]string{
+				"app.kubernetes.io/name":       "prometheus-agent",
+				"app.kubernetes.io/instance":   name,
+				"app.kubernetes.io/managed-by": "prometheus-operator",
+			},
+		},
+	}
+
+	_, err := framework.KubeClient.CoreV1().Services(ns).Create(context.Background(), svc, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	framework.SetupPrometheusRBAC(context.Background(), t, testCtx, ns)
+
+	p := framework.MakeBasicPrometheusAgent(ns, name, name, 1)
+	p.Spec.ServiceName = ptr.To(fmt.Sprintf("%s-service", name))
+
+	_, err = framework.CreatePrometheusAgentAndWaitUntilReady(context.Background(), ns, p)
+	require.NoError(t, err)
+
+	// The Agent doesn't support querying. So we just ensure that we are able to reach the
+	// Prometheus Agent through our service.
+	targets, err := framework.GetActiveTargets(context.Background(), ns, svc.Name)
+	require.NoError(t, err)
+	require.Empty(t, targets)
+
+	// Ensure that governing service was not created.
+	governingServiceName := "prometheus-operated"
```

Comment:
it might be more reliable to list all the services in the namespace and check that we have only one item (same comment for the prometheus test).
```suggestion
	governingServiceName := "prometheus-agent-operated"
```

Reply:
Will check the number of services instead.

Reply:
And also added a check to ensure that the one service we have is the one we expect. This should be pretty reliable.